### PR TITLE
some improvements to the group profile card

### PR DESCRIPTION
**Pull request by punkle**
## Hey, I just made a Pull Request!

some improvements to the group profile card:

- The card title will now allow the use of the entity.metadata.title. Now it will either use the profile.displayName, metadata.title, and then the entity name which ever comes first. Previously it did not use the title.
- The group profile card also now has an optional setting to enable a peek ahead popover which shows the details of the parent and child groups.
- The entity ref link/s now allow an optional prop to enable the peek ahead popover.

Entity title used as the header of the card:
image

Peek ahead on the parent groups:
image

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by punkle** [comment]
@Edje-C I added storybook items and added screenshots.

**Comment by punkle** [comment]
I dont think the failure is related to my change.

File: plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx [hunk 1]
```diff
@@ -46,7 +49,14 @@ export type EntityRefLinkProps = {
  */
 export const EntityRefLink = forwardRef<any, EntityRefLinkProps>(
   (props, ref) => {
-    const { entityRef, defaultKind, title, children, ...linkProps } = props;
+    const {
+      entityRef,
+      usePeekAheadPopover = false,
```
Comment (Edje-C) [inline plugins/catalog-react/src/components/EntityRefLink/EntityRefLink.tsx:54]: I don't think this is necessary. It's either explicitly true or false or undefined which are both falsy.

File: plugins/catalog-react/src/components/EntityRefLink/EntityRefLinks.tsx [hunk 2]
```diff
@@ -50,8 +52,14 @@ export type EntityRefLinksProps<
 export function EntityRefLinks<
   TRef extends string | CompoundEntityRef | Entity,
 >(props: EntityRefLinksProps<TRef>) {
-  const { entityRefs, defaultKind, fetchEntities, getTitle, ...linkProps } =
-    props;
+  const {
+    entityRefs,
+    usePeekAheadPopover = false,
```
Comment (Edje-C) [inline plugins/catalog-react/src/components/EntityRefLink/EntityRefLinks.tsx:57]: again here

File: plugins/catalog-react/src/components/EntityRefLink/FetchedEntityRefLinks.tsx [hunk 3]
```diff
@@ -47,7 +48,13 @@ export type FetchedEntityRefLinksProps<
 export function FetchedEntityRefLinks<
   TRef extends string | CompoundEntityRef | Entity,
 >(props: FetchedEntityRefLinksProps<TRef>) {
-  const { entityRefs, defaultKind, getTitle, ...linkProps } = props;
+  const {
+    entityRefs,
+    usePeekAheadPopover = false,
```
Comment (Edje-C) [inline plugins/catalog-react/src/components/EntityRefLink/FetchedEntityRefLinks.tsx:53]: again here

**Review by Edje-C — COMMENTED** [review]
We're not looking to accept PRs, especially UI changes, without some discussion on the need /broader desire for such updates first. Is this work the product of an issue or conversation?

**Comment by punkle** [comment]
closes #16961

**Comment by punkle** [comment]
closes #16962

**Comment by punkle** [comment]
@Edje-C I hadnt created issues. Ive created them and linked them https://github.com/backstage/backstage/issues/16961 and https://github.com/backstage/backstage/issues/16962

**Comment by Edje-C** [comment]
Hi, we're discussing internally how we want to handle `EntityPeekAheadPopover` usage. I'll get back to you with any updates.

**Comment by Edje-C** [comment]
Awesome 😂
Thank you so much!

**Comment by punkle** [comment]
@Edje-C It would be great to hear an update about the discussions when they take place? Also I am curious, when you say internal, what do you mean?

**Comment by Edje-C** [comment]
> @Edje-C It would be great to hear an update about the discussions when they take place? Also I am curious, when you say internal, what do you mean?

Internally as in amongst maintainers. 
The discussion has been that this seems like another reason to take a look at getting some composability patterns in place because essentially that's the best way to solve this. So we'll be bringing that conversation up in the SIG. 
https://github.com/backstage/backstage/issues/16876

We're looking at proposing some middle ground to solve for this specific use case in the mean time though, so we'll be back with that sometime soon.

**Comment by punkle** [comment]
not stale

**Comment by punkle** [comment]
not stale

**Comment by Edje-C** [comment]
Hello
An update: @freben is working on an in depth solution to the issue you're trying to solve here. It was discussed it in the SIG today and when the PRFC is out, we'll tag you directly for feedback.

**Comment by punkle** [comment]
ok, Ill close this in the meantime.

**Comment by Edje-C** [comment]
I should've mentioned this but here's a link to what was discussed about his approach
https://docs.google.com/document/d/1xg57CDfX2R5jTGbbWANzifm2szEb664kuTwfEEvKgZU/edit#bookmark=id.t2dm0zwlfhqb

**Pull request closed without merging**